spec/support/pages/portfolios/index.rb

```diff
@@ -0,0 +1,145 @@
+
+require "support/pages/page"
+
+module Pages
+  module Portfolios
+    class Index < ::Pages::Page
+      include ::Components::Common::Filters
+      include ::Components::Autocompleter::NgSelectAutocompleteHelpers
+
+      def path(*)
+        "/portfolios"
+      end
+
+      def expect_portfolios_listed(*portfolios)
+        within_portfolio_list do
+          portfolios.each do |portfolio|
+            expect(page).to have_text(portfolio.name)
+            expect(page).to have_text(portfolio.description)
+          end
+        end
+      end
+
+      def expect_portfolios_not_listed(*portfolios)
+        within_portfolio_list do
+          portfolios.each do |portfolio|
+            case portfolio
+            when Project
+              expect(page).to have_no_text(portfolio.name)
+            when String
+              expect(page).to have_no_text(portfolio)
+            else
+              raise ArgumentError, "#{portfolio.inspect} is not a Portfolio or a String"
+            end
+          end
+        end
+      end
+
+      def expect_portfolio_at_place(portfolio, place)
+        within_portfolio_list do
+          expect(page)
+            .to have_css(".portfolio:nth-of-type(#{place}) .portfolio-name", text: portfolio.name)
+        end
+      end
+
+      def expect_portfolios_in_order(*portfolios)
+        portfolios.each_with_index do |portfolio, index|
+          expect_portfolio_at_place(portfolio, index + 1)
+        end
+      end
+
+      def expect_title(name)
+        expect(page).to have_css('[data-test-selector="portfolio-query-name"]', text: name)
+      end
+
+      def filter_by_active(value)
+        set_filter("active", "Active", "is", [value])
+        wait_for_reload
+      end
+
+      def filter_by_name_and_identifier(value, send_keys: false)
+        set_name_and_identifier_filter([value], send_keys:)
+        wait_for_reload
+      end
+
+      def open_more_menu
+        wait_for_network_idle
+        page.find('[data-test-selector="portfolio-more-dropdown-menu"]').click
+      end
+
+      def expect_more_menu_item(item)
+        open_more_menu
+        expect(page).to have_css(".ActionListItem", text: item, exact_text: true)
+      end
+
+      def click_more_menu_item(item)
+        open_more_menu
+        page.find(".ActionListItem", text: item, exact_text: true).click
+        wait_for_network_idle
+      end
+
+      def expect_new_portfolio_button
+        expect(page).to have_css('[data-test-selector="portfolio-new-button"]')
+      end
+
+      def expect_no_new_portfolio_button
+        expect(page).to have_no_css('[data-test-selector="portfolio-new-button"]')
+      end
+
+      def create_new_portfolio
+        page.find('[data-test-selector="portfolio-new-button"]').click
+      end
+
+      def sidebar_menu_items
+        page.find_by_id("menu-sidebar").all(".op-submenu--item-title").map(&:text)
+      end
+
+      def within_portfolio_list(&)
+        within "#portfolios-index-container", &
```

Comment:
You could use the test selector here via `within_test_selector` and by that wouldn't need to have an id to begin with.